packages/inter-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -564,7 +559,7 @@ export const prepareVaultManagerKit = (
             totalShortfallReceived: state.totalShortfallReceived,
           });
 
-          metricsPublisher.publish(payload);
+          return E(metricsTopicKit.recorderP).write(payload);
```

Comment:
I'm not convinced that a synchronous `assetKit` is beneficial, given the necessarily asynchronous `metricsTopicKit` (e.g., use of `makeERecorderKit` for both seems better to me). But I'm not so certain of this that I consider it a blocker, so I'll defer to your judgment on it.